tests_v2/test_workspace_icon_actions.py

```
@@ -0,0 +1,112 @@
+        return json.loads(proc.stdout or "{}")
+    except Exception as exc:
+        raise cmuxError(f"Invalid JSON output for {' '.join(args)}: {proc.stdout!r} ({exc})")
+
+
+def _workspace_icon(c: cmux, workspace_id: str) -> Dict[str, Any] | None:
+    rows = (c._call("workspace.list", {}) or {}).get("workspaces") or []
+    for row in rows:
+        if str(row.get("id") or "") == workspace_id:
+            icon = row.get("custom_icon")
+            return icon if isinstance(icon, dict) else None
+    raise cmuxError(f"workspace.list did not include workspace {workspace_id}")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)